packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -502,63 +503,24 @@ abstract class NetworkImage extends ImageProvider<NetworkImage> {
 /// Decodes the given [File] object as an image, associating it with the given
 /// scale.
 ///
+/// This class is not supported when build Flutter for web applications.
+///
 /// See also:
 ///
 ///  * [Image.file] for a shorthand of an [Image] widget backed by [FileImage].
-class FileImage extends ImageProvider<FileImage> {
+abstract class FileImage implements ImageProvider<FileImage> {
   /// Creates an object that decodes a [File] as an image.
   ///
+  /// [file] must be an instance of the `dart:io` [File] type.
+  ///
   /// The arguments must not be null.
-  const FileImage(this.file, { this.scale = 1.0 })
-    : assert(file != null),
-      assert(scale != null);
+  const factory FileImage(Object file, { double scale }) = file_image.FileImage;
 
   /// The file to decode into an image.
-  final File file;
+  final Object file;
```

Comment:
Could we come up with a foundation type for this instead of Object?  Maybe for a future PR.

Reply:
I think anything else (besides dynamic) is actually a breaking change. If the type implements File, then dart:io is still a dependency of the API. If it doesn't, then it breaks callsites.

If we decide to make a breaking change, it would be easier to maintain this API if it took a path String instead.